<launch>


  <!-- Node that republishes vicon info to the input format of the
       estimation node -->
  <node
       name="read_publish_object_state"
       pkg="ros_pybullet_interface"
       type="read_publish_object_estimation.py"
       clear_params="true"
       output="screen">

       <param
           name="sim_or_real"
           value="real"
           type="str"/>

  </node>


  <!-- Flag that activates forwarding of commands to the robots -->
  <arg name="command_real_robots" default="false"/>

  <node
      name="sim_HW_republisher_yang"
      pkg="ros_pybullet_interface"
      type="sim_joint_cmd_stiffness_2_FRI.py"
      output="screen">

    <param
        name="robot_name"
        value="yang"
        type="str"/>

    <param
        name="cmd_real_robot"
        value="$(arg command_real_robots)"
        type="bool"/>

  </node>

  <node
      name="sim_HW_republisher_yin"
      pkg="ros_pybullet_interface"
      type="sim_joint_cmd_stiffness_2_FRI.py"
      output="screen">

    <param
        name="robot_name"
        value="yin"
        type="str"/>

    <param
        name="cmd_real_robot"
        value="$(arg command_real_robots)"
        type="bool"/>

  </node>

  <!-- PyRBDL4dIK Environmet node (receiver - publisher) -->
  <node
      name="ros_PyRBDL4dIK_interface"
      pkg="ros_pybullet_interface"
      type="ros_rbdl_IK_interace_node.py"
      output="screen">

    <param
        name="robot_config"
        value="{ros_pybullet_interface}/configs/yin_visual.yaml"
        type="str"/>


    <!-- remap such that the initial state of the IK comes from real world state -->
    <remap if="$(arg command_real_robots)" from="yin_visual/ros_pybullet_interface/joint_state/current"  to="yin/ros_pybullet_interface/joint_state/target" />


  </node>


  <node
      name="ros_PyRBDL4dIK_interface2"
      pkg="ros_pybullet_interface"
      type="ros_rbdl_IK_interace_node.py"
      output="screen">

    <param
        name="robot_config"
        value="{ros_pybullet_interface}/configs/yang_visual.yaml"
        type="str"/>


        <!-- remap such that the initial state of the IK comes from real world state -->
        <remap if="$(arg command_real_robots)" from="yang_visual/ros_pybullet_interface/joint_state/current"  to="yang/ros_pybullet_interface/joint_state/target" />


  </node>



</launch>
